pkg/controllers/management/eksupstreamrefresh/eks_upstream_refresh.go

```diff
@@ -95,14 +98,26 @@ func (e *eksRefreshController) refreshClusterUpstreamSpec(cluster *mgmtv3.Cluste
 	}
 
 	upstreamSpec, err := GetComparableUpstreamSpec(e.secretsCache, cluster)
-	if err != nil {
+	if err != nil && upstreamSpec == nil {
+		var statusErr error
+		cluster = cluster.DeepCopy()
+		apimgmtv3.ClusterConditionUpdated.False(cluster)
+		apimgmtv3.ClusterConditionUpdated.Message(cluster, "[Syncing error: syncing broken] "+err.Error())
+		cluster, statusErr = e.clusterClient.Update(cluster)
+		if statusErr != nil {
+			return cluster, statusErr
+		}
 		return cluster, err
 	}
 
 	if !reflect.DeepEqual(cluster.Status.EKSStatus.UpstreamSpec, upstreamSpec) {
 		logrus.Infof("updating cluster [%s], upstream change detected", cluster.Name)
 		cluster = cluster.DeepCopy()
 		cluster.Status.EKSStatus.UpstreamSpec = upstreamSpec
+		if err != nil {
```

Comment:
This section is quite confusing to me, is this err from `GetComparableUpstreamSpec` but upstreamSpec isn't nil and it's not equal ? Can we push this up outside of the conditional ?

Reply:
If `err != nil && upstreamSpec == nil`, then the syncing failed for some reason, so it is reported to the user in the Updated message.

If `upstreamSpec != nil`, then the syncing occurred as expected. If `err != nil` in this case, then there were health issues with one (or more) of the node groups that needs to be reported to the user. These are health issues like "the node group is degraded because the user deleted the launch template." These types of health issues are not reported to the user at any other time because there are not Kubernetes health issues.

Reply:
My question is why wouldn't this be on line 116, as just `if err != nil && upstreamSpec != nil`. In this block we are also checking if its the same as spec. So where is this case being handled ?

Reply:
Ah yes. I believe I understand what you are saying, and this does need to be moved.